c7n/resources/s3.py

```diff
@@ -2494,3 +2500,119 @@ def process_bucket(self, bucket):
+                resource: s3
+                actions:
+                  - type: set-bucket-encryption
+                  # enabled: true <------ optional (true by default)
+                    crypto: aws:kms
+                    key: 1234abcd-12ab-34cd-56ef-1234567890ab
+
+              - name: s3-enable-default-encryption-kms-alias
+                resource: s3
+                actions:
+                  - type: set-bucket-encryption
+                  # enabled: true <------ optional (true by default)
+                    crypto: aws:kms
+                    key: alias/some/alias/key
+
+              - name: s3-enable-default-encryption-aes256
+                resource: s3
+                actions:
+                  - type: set-bucket-encryption
+                  # crypto: AES256 <----- optional (AES256 by default)
+                  # enabled: true <------ optional (true by default)
+
+              - name: s3-disable-default-encryption
+                resource: s3
+                actions:
+                  - type: set-bucket-encryption
+                    enabled: false
+    """
+
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'properties': {
+            'type': {'enum': ['set-bucket-encryption']},
+            'enabled': {'type': 'boolean'},
+            'crypto': {'enum': ['aws:kms', 'AES256']},
+            'key': {'type': 'string'}
+        },
+        'dependencies': {
+            'key': {
+                'properties': {
+                    'crypto': {'pattern': 'aws:kms'}
+                },
+                'required': ['crypto']
+            }
+        }
+    }
+
+    permissions = ('s3:PutEncryptionConfiguration',
+                's3:GetEncryptionConfiguration',
+                'kms:ListAliases')
+
+    def process(self, buckets):
+        with self.executor_factory(max_workers=3) as w:
+            results = []
+            enabled = self.data.get('enabled', True)
+            if enabled:
+                SSEAlgorithm = self.data.get('crypto', 'AES256')
+                config = {'Rules': [
+                    {
+                        'ApplyServerSideEncryptionByDefault': {
+                            'SSEAlgorithm': SSEAlgorithm
+                        }
+                    }
+                ]}
+                key = self.data.get('key')
+                if key:
+                    config['Rules'][0]['ApplyServerSideEncryptionByDefault']['KMSMasterKeyID'] = key
+            else:
+                config = None
+            futures = {w.submit(self.process_bucket, b, config) : b for b in buckets}
+            for future in as_completed(futures):
+                if future.exception():
+                    bucket = futures[future]
+                    self.log.error('error enabling bucket encryption: %s\n%s',
+                                   bucket['Name'], future.exception())
+                results += filter(None, [future.result()])
```

Comment:
given process bucket never returns a value this will always be empty and is therefore extraneous.